samcli/commands/validate/validate.py

```diff
@@ -146,12 +147,18 @@ def _lint(ctx: Context, template: str) -> None:
         matches = list(cfnlint.core.get_matches(filenames, args))
         if not matches:
             click.secho("{} is a valid SAM Template".format(template), fg="green")
+            return
+
         rules = cfnlint.core.get_used_rules()
         matches_output = formatter.print_matches(matches, rules, filenames)
 
         if matches_output:
             click.secho(matches_output)
 
+        raise LinterRuleMatchedException(
```

Comment:
If there are 0 matches, do we know that we are in exit code 1 state by default?

Reply:
The opposite actually. Line 148 checks for any matches and logs a success message if none are found
```
if not matches:
     click.secho("{} is a valid SAM Template".format(template), fg="green")
```
Otherwise there are some rules matched (meaning the linter found some issues) and we're in exit code 1 state.